### ISSUE TITLE
Increase text contrast in Material demos

### ISSUE THREAD

**Issue by jonahwilliams** [flutter/flutter#21651]
Search demo:

Contrast ratio is currently 3.12 but should be at least 4.5. I believe the fab should use the 500 shade instead of the accent shade.

Foreground color: #FFFFFF
Background color:  #2196F3

**Comment by jonahwilliams** [comment]
Edit: the "Share" button in the cards demo also has insufficient contrast.

**Comment by SUPERETDUPER** [comment]
I can contribute to this issue.

**Comment by SUPERETDUPER** [comment]
Looking at the material design guidlines, accent color (shade 200) is often used for FAB. Maybe the text color needs to be inverted. I'm trying to determine the precise color of the button but for now the flutter inspector isn't working.

**Comment by jonahwilliams** [comment]
Hey @SUPERETDUPER 

Unfortunately the material design guidelines which recommend accent color contradict the accessibility guidelines for text contrast. The primary (500) shade has to be used instead for things like fabs and buttons.

If you're interested in contributing here, take a look at the tests in https://github.com/flutter/flutter/blob/master/examples/flutter_gallery/test/accessibility_test.dart that are being skipped.
